apps/nextjs/src/app/page.tsx

```diff
@@ -33,10 +32,21 @@ export default async function HomePage() {
               </div>
             }
           >
-            <PostList posts={posts} />
+            <PostListWrapped />
           </Suspense>
         </div>
       </div>
     </main>
   );
 }
+
+async function PostListWrapped() {
+  const posts = await api.post.all();
```

Comment:
1. Maybe the result of `api.post.all()` could be _branded_ somehow so we know the or can figure out the full query key